src/app/add-item/add-item.component.ts

```diff
@@ -12,22 +12,36 @@ import { first } from 'rxjs/operators';
 })
 export class AddItemComponent implements OnInit {
```

Comment:
**SUGGESTION:** As it's not only adding now but also updating then maybe it'd be worth to name it eg. `ManageActionItemComponent` or `AddOrUpdateActionItemComponent`